docs/README.md

```diff
@@ -9,36 +9,80 @@ unreleased features.
 ## Building the docs
 
 ### Setup
-1. Install LanceDB. From LanceDB repo root: `pip install -e python`
-2. Install dependencies. From LanceDB repo root: `pip install -r docs/requirements.txt`
-3. Make sure you have node and npm setup
-4. Make sure protobuf and libssl are installed
+1. Install LanceDB Python. See setup in [Python contributing guide](../python/CONTRIBUTING.md).
+   Run `make develop` to install the Python package.
+2. Install documentation dependencies. From LanceDB repo root: `pip install -r docs/requirements.txt`
 
-### Building node module and create markdown files
+### Preview the docs
 
-See [Javascript docs README](./src/javascript/README.md)
+```shell
+cd docs
+mkdocs serve
+```
 
-### Build docs
-From LanceDB repo root:
+If you want to just generate the HTML files:
 
-Run: `PYTHONPATH=. mkdocs build -f docs/mkdocs.yml`
+```shell
+PYTHONPATH=. mkdocs build -f docs/mkdocs.yml
+```
 
 If successful, you should see a `docs/site` directory that you can verify locally.
 
-### Run local server
+## Adding examples
 
-You can run a local server to test the docs prior to deployment by navigating to the `docs` directory and running the following command:
+To make sure examples are correct, we put examples in test files so they can be
+run as part of our test suites.
 
-```bash
-cd docs
-mkdocs serve
+You can see the tests are at:
+
+* Python: `python/python/tests/docs`
+* Typescript: `nodejs/examples/`
+
+### Checking python examples
+
+```shell
+cd python
+pytest -vv python/tests/docs
 ```
 
-### Run doctest for typescript example
+### Checking typescript examples
 
-```bash
-cd lancedb/docs
-npm i
+The `@lancedb/lancedb` package must be built before running the tests:
+
+```shell
+pushd nodejs
+npm run install
 npm run build
-npm run all
+popd
+```
+
+Then you can run the examples by going to the `nodejs/examples` directory and
+running the tests like a normal npm package:
+
+```shell
+pushd nodejs/examples
+npm install
+npm test
+popd
+```
+
+## API documentation
+
+### Python
+
+The Python API documentation is organized based on the file `docs/src/python/python.md`.
+We manually add entries there so we can control the organization of the reference page.
+**However, this means any new APIs must be manually added to the file.** No additional
+steps are needed to generate the API documentation.
+
+### Typescript
+
+The typescript API documentation is generated from the typescript source code using [typedoc](https://typedoc.org/).
+
+When new APIs are added, you must manually re-run the typedoc command to update the API documentation.
```

Comment:
Do these files need to be committed and checked in?  If so we should mention that explicitly.

Reply:
That's a good point!